flocker/control/_model.py

```diff
@@ -723,35 +757,35 @@ def move_application(self, application, target_node):
         """
         deployment = self
         for node in deployment.nodes:
-            for container in node.applications:
```

Comment:
This is the (or part of the) crux of this branch, right?

We no longer have to loop through the entire set. Just lookup the application by name.

Cool.